### ISSUE TITLE
Documentation: Basics.md somewhat misleading..

### ISSUE THREAD

**Issue by fps** [FluxML/Flux.jl#305]
The first section "Taking Gradients" introduces a simple linear model and takes the reader through the process of taking the gradient of the loss function with respect to the parameters and even takes a single step of gradient descent. 

There is an obvious extrapolation of this section to performing the calculation of the loss and its gradient and performing the updating of the parameters prepeatedly in a loop like e.g. this:

<pre>
for n=1:00
  l = loss(x, y)
  println("Loss: ", l)
  back!(l)
  W.data .-= 0.1(W.grad)
end
</pre>

This "obvious" extrapolation fails to work though. It seems one needs to clear the gradients after each update of the parameters and reapplication of the loss function.

My suggestion would be to expand the section a little bit making this point clear. I only mention this because I lost a day or two before I realized I should take a closer look at the calculated gradients and how they made no sense at all :) The section could then end with a hint that the built in optimizers in Flux do all this grunt work for the user..

**Comment by MikeInnes** [comment]
Yeah, I can see why that'd be confusing – thanks for raising it.

It's only a small patch, but what do you think of https://github.com/FluxML/Flux.jl/commit/bed6d2311e4c9778c8057249ccce9a636f334cc0? I think it'd avoid someone being your position again.

**Comment by fps** [comment]
Yeah, I suppose that would have helped :)
